### PR TITLE
Process creating NcML file aggregating netCDF files

### PR DESCRIPTION
**Pull request by huard**
## Overview

This process creates two netCDF files, and an NcML aggregating those files. This is intended to be installed on a test environment, so that when the process is called, files are created and picked up by the THREDDS server. Clients can then test that the NcML endpoint works as expected.

### CODE REVIEW

**Comment by huard** [comment]
@cehbrecht Can this be useful in the context of the CDS and Thursday's discussions ?

File: emu/processes/wps_test_ncml.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+from pywps.app.Common import Metadata
+from pathlib import Path
+
+from netCDF4 import Dataset
+
+import logging
+LOGGER = logging.getLogger("PYWPS")
+
+
+class NcMLTest(Process):
```
Comment (cehbrecht) [inline emu/processes/wps_test_ncml.py]: Where is the difference to NcMLAgg?

**Comment by cehbrecht** [comment]
@huard You can preset this as an example for ncml usage in the discussion on Thursday. Maybe you like also add an example with ncml where the metadata of the files gets "fixed"?

I have put my thoughts for the discussion here:
https://github.com/cehbrecht/climate-operators-prototype/blob/master/README.md

Looking at the ESMValTool we expect *fixes* that can not be done with ncml. Also the ESGF Future Architecture might not use Thredds/OpenDAP anymore. Just plain file downloading using an Nginx web service. I thought one could use FUSE (File System in User Space) to provide a virtual data layer as a file-system. The FUSE could apply the data fixes on top of the original data when data gets requested. Probably a caching is wanted here as well. But I can not say if this is possible and reasonable ...

File: emu/processes/wps_test_ncml.py (hunk 1 above)
Comment (huard) [reply emu/processes/wps_test_ncml.py]: None, this is a left-over from a file rename...

**Comment by huard** [comment]
Will do, this was just a smoke test. We still need to figure out the client part of the test. 

Is Emu the right place to put these types of test processes? I could maintain an Emu branch with those processes if you feel that it will confuse users.

**Comment by cehbrecht** [comment]
> Is Emu the right place to put these types of test processes?

I think Emu is ok. We can move them to other places later if we have the feeling they belong somewhere else.

**Comment by huard** [comment]
So the process currently returns an URL, but for NcML we'll need an opendap link.

**Comment by huard** [comment]
Requires https://github.com/geopython/pywps/pull/515 and https://github.com/geopython/pywps/pull/513

File: emu/processes/wps_nc_to_dap.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            status_supported=True)
+
+    def _handler(self, request, response):
+        url = request.inputs['resource'][0].url
+
+        # Write response
+        file_server = configuration.CONFIG.get("server", "outputurl")
+        dap_server = configuration.CONFIG.get("dap", "outputurl")
+
+        response.outputs["dap"].url = url.replace(file_server, dap_server)
```
Comment (cehbrecht) [inline emu/processes/wps_nc_to_dap.py]: should not the `DapStorageBuilder` handle the storage of the input netcdf file?

Comment (huard) [reply emu/processes/wps_nc_to_dap.py]: It's probably good to have a backup mechanism to do this in the case a process is very long to run, and you don't want to run it again just to change the url. 
Another option would be to have a cache in pywps returning a previously computed value if all process input parameters are identical, but that's not trivial.

Comment (cehbrecht) [reply emu/processes/wps_nc_to_dap.py]: Is this one obsolete with the configuration change for OpenDAP?

Comment (huard) [reply emu/processes/wps_nc_to_dap.py]: I think so.

**Comment by huard** [comment]
@cehbrecht I think this is ready to merge.

**Pull request merged**